backend/lcfs/web/api/user/views.py

```diff
@@ -44,100 +45,164 @@ async def get_current_user(request: Request, response: Response = None) -> UserB
             err_msg = "Current user not found"
             logger.error(err_msg)
             response.status_code = status.HTTP_404_NOT_FOUND
-            return UserBase(status=status.HTTP_404_NOT_FOUND,
-                                  message="Not Found", success=False, data={},
-                                  error={"message": err_msg})
-        return UserBase.from_orm(current_user)
+            return UserBase(
+                status=status.HTTP_404_NOT_FOUND,
+                message="Not Found",
+                success=False,
+                data={},
+                error={"message": err_msg},
+            )
+        return UserBase.model_validate(current_user)
     except Exception as e:
-        response.status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
         logger.error("Error getting current user", str(e))
-        return EntityResponse(status=status.HTTP_500_INTERNAL_SERVER_ERROR,
-                              message="Failed", success=False, data={},
-                              error={"message": f"Technical error: {e}"})
-
+        return UserBase(
+            status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            message="Failed",
+            success=False,
+            data={},
+            error={"message": f"Technical error: {e.args[0]}"},
+        )
 
 
-@router.get("/", response_model=List[UserBase], status_code=status.HTTP_200_OK)
+@router.post("", response_model=EntityResponse, status_code=status.HTTP_200_OK)
 @roles_required("Government")
-async def get_users(limit: int = 10, offset: int = 0, response: Response = None) -> List[UserBase]:
-    current_page = math.ceil(offset / limit) + 1
+async def get_users(
+    request: Request,
+    pagination: PaginationRequestSchema = Body(..., embed=False),
+    response: Response = None,
+) -> List[UserBase]:
     try:
-        users = await user_repo.query_users(
-            pagination={
-                "per_page": limit, 
-                "page": offset, 
-                "sort_field": "username", 
-                "sort_direction": "asc"
-            }
-        )
+        users, total_count = await user_repo.query_users(pagination=pagination)
+
         if len(users) == 0:
             logger.error("Error getting users")
             response.status_code = status.HTTP_404_NOT_FOUND
-            return []
-        
-        # Convert the users to a list of UserBase objects
-        user_bases = [UserBase.from_orm(user) for user in users]
-        return user_bases  # Return the list of UserBase objects
+            return EntityResponse(
+                status=status.HTTP_404_NOT_FOUND,
+                message="Failed",
+                success=False,
+                data=[],
+                error={"message": "No roles found"},
+            )
+
+        # Return the list of UserBase objects
+        return EntityResponse(
```

Comment:
will we always use EntityResponse for these pagination responses?

Reply:
If it's only pagination then we wouldn't need it, as it involves sort and filters this is more efficient I felt. If not we could go with any 3rd party libraries.